### PR TITLE
Update vagrant default

### PR DESCRIPTION
**Pull request by rstyd**
This PR addresses #511. It updates the default `vagrant up` behavior so that it uses the newest tagged version of charliecloud and updates the documentation to reflect that.

### CODE REVIEW

File: doc-src/vm.rst [hunk 1]
```diff
@@ -182,7 +182,7 @@ Build and provision
 To build the VM and install Docker, Charliecloud, etc.::
 
   $ cd packaging/vagrant
-  $ CH_VERSION=v0.9.1 vagrant up
+  vagrant up
```
Comment (reidpr) [inline doc-src/vm.rst]: Prompt (`$`) got lost here and below.

File: packaging/vagrant/Vagrantfile [hunk 2]
```diff
@@ -182,9 +182,11 @@ EOF2
     git clone --recursive https://github.com/hpc/charliecloud.git
 
     cd charliecloud
-    if [[ $CH_VERSION ]]; then
-      git checkout $CH_VERSION
+    if [[ -z $CH_VERSION ]]; then
+      CH_VERSION=$(git tag --sort=-version:refname --format='%(refname:strip=2)' | head -n 1)
     fi
+    git checkout $CH_VERSION
+
```
Comment (reidpr) [inline packaging/vagrant/Vagrantfile]: Spurious newline added?

File: packaging/vagrant/Vagrantfile [hunk 3]
```diff
@@ -182,9 +182,11 @@ EOF2
     git clone --recursive https://github.com/hpc/charliecloud.git
 
     cd charliecloud
-    if [[ $CH_VERSION ]]; then
-      git checkout $CH_VERSION
+    if [[ -z $CH_VERSION ]]; then
+      CH_VERSION=$(git tag --sort=-version:refname --format='%(refname:strip=2)' | head -n 1)
     fi
+    git checkout $CH_VERSION
```
Comment (reidpr) [inline packaging/vagrant/Vagrantfile]: Would it be useful to print what version of Charliecloud we found somewhere?

File: doc-src/vm.rst [hunk 4]
```diff
@@ -182,7 +182,7 @@ Build and provision
 To build the VM and install Docker, Charliecloud, etc.::
 
   $ cd packaging/vagrant
-  $ CH_VERSION=v0.9.1 vagrant up
```
Comment (reidpr) [inline doc-src/vm.rst]: Looks like the meaning of `CH_VERSION` needs to be updated a few lines below.

File: packaging/vagrant/Vagrantfile (hunk 3 above)
Comment (rstyd) [reply packaging/vagrant/Vagrantfile]: I think so.

File: packaging/vagrant/Vagrantfile (hunk 2 above)
Comment (rstyd) [reply packaging/vagrant/Vagrantfile]: Yes! Good catch.

File: doc-src/vm.rst (hunk 4 above)
Comment (rstyd) [reply doc-src/vm.rst]: Should we have a way for people to pull the head of the master branch?

**Comment by rstyd** [comment]
I agree. I like all the tidying.

**Comment by reidpr** [comment]
OK, those two bugs are merged. Please revisit this issue and see if the tests pass.

**Pull request merged**